src/manager.cr

```diff
@@ -46,7 +58,8 @@ class HTTPSession
     end
 
     def set(context : HTTP::Server::Context, session : T) : Nil
-      unless session_id = session_id(context)
+      session_id = session_id(context)
+      unless session_id && @storage[session_id]
         session_id = new_session_id
```

Comment:
I'm a bit hesitant about this behavior. If the session ID is invalid I may want to know about that (to return a 400 perhaps?), so maybe you want to provide `#set` and `#set!`?

Reply:
Yeah, that's a good point. Even if not making it an error, it could at least be something you might want to put in a log of suspicious activity.
That's even mentioned here: https://cheatsheetseries.owasp.org/cheatsheets/Session_Management_Cheat_Sheet.html#session-id-generation-and-verification-permissive-and-strict-session-management 🙈